run

```diff
@@ -656,6 +656,127 @@ copy_data_() {
     run_command cp -r ${SRC_ROOT}/3rdparty ${SRC_ROOT}/LICENSE* ${DEST_ROOT}/
 }
 
+
+#==================================================================================
+# Section: Test
+#==================================================================================
+
+test_desc() { echo 'Execute test cases
+
+Arguments:
+  $1 - subcommand [all|python|c++] (default: all)
+  $2 - test_type [all|unit|integration|system|performance] (default: all)
+  $3 - test_component [all|clara|skimage] (default: all)
+'
+}
+test() {
+    local subcommand="${1:-all}"
+    local test_type="${2:-all}"
+    shift;
+
+    if [ "$subcommand" = "all" ] || [ "$subcommand" = "python" ]; then
+        test_python "$@"
+    fi
+}
+
+check_install_python_dep_() {
+    local package_names="$@"
+
+    [ -z "$package_names" ] && c_echo_err R "Please specify package names to check!" && return 1
+
+    local packages_to_install=""
+    local has_all_packages=true
+
+    for package_name in $package_names; do
+        # `package_name` has '<package name>[:<conda package name>]' format.
+        if [ -n "${CONDA_PREFIX}" ]; then
+            # abc:def => def
+            # abc => abc
+            package_name="${package_name##*:}"
+            if ! conda list | grep -i -c "${package_name} " > /dev/null; then
+                c_echo G "${package_name}" W " doesn't exists. Installing " G "${package_name}" W "..."
+                has_all_packages=false
+
+                # Workaround for `openslide-python`
+                # (See https://github.com/conda-forge/openslide-python-feedstock/issues/4)
+                [ "${package_name}" = "openslide-python" ] && package_name="conda-forge/linux-64::openslide-python"
+
+                packages_to_install="${packages_to_install} ${package_name}"
+            fi
+        else
+            # abc:def => abc
+            # abc => abc
+            package_name="${package_name%%:*}"
+            if ! pip3 list | grep -i -c "${package_name} " > /dev/null; then
+                c_echo G "${package_name}" W " doesn't exists. Installing " G "${package_name}" W "..."
+                has_all_packages=false
+                packages_to_install="${packages_to_install} ${package_name}"
+            fi
+        fi
+    done
+    if [ "$has_all_packages" != "true" ]; then
+        if [ -n "${CONDA_PREFIX}" ]; then
+            run_command conda install -c conda-forge -y ${packages_to_install}
+        else
+            if [ -n "${VIRTUAL_ENV}" ]; then
+                run_command pip3 install ${packages_to_install}
+            else
+                run_command pip3 install --user ${packages_to_install}
+            fi
+        fi
+    fi
+    hash -r
+}
+
+install_python_test_deps_() {
+    # Package name has '<package name>[:<conda package name>]' format.
+    check_install_python_dep_ pytest pytest-cov pytest-lazy-fixture psutil GPUtil tifffile imagecodecs openslide-python
```

Comment:
may want to keep these in a separate `requirements-test.txt` file or similar? If this is the only place they are used, then it is probably not a big deal either way.

Reply:
That's good idea! Will also use fixed versions of the packages.